akka-stream/src/main/scala/akka/stream/impl/fusing/Ops.scala

```diff
@@ -1846,9 +1846,18 @@ private[stream] object Collect {
 
   override def createLogic(inheritedAttributes: Attributes): GraphStageLogic =
     new TimerGraphStageLogic(shape) with InHandler with OutHandler {
-      def onPush(): Unit = push(out, grab(in))
+      private val timeoutNs = System.nanoTime() + timeout.toNanos
 
-      def onPull(): Unit = pull(in)
+      def overdue() = System.nanoTime() >= timeoutNs
```

Comment:
Then there will be so much system call.

Reply:
Yeah, that's the downside of it. One guard per element. Perhaps we could toggle it and only use it when the timeout is very short (which should anyway limit the number of elements).

Reply:
The new way is OK:)